code/game/gamemodes/cult/runes.dm

```diff
@@ -1,1053 +1,983 @@
-	"<span class='danger'>The blood starts flowing from the rune and into your frail mortal body. You feel... empowered.</span>", \
-	"<span class='italics'>You hear a liquid flowing.</span>")
-	var/mob/living/user = usr
-	if(user.bhunger)
-		user.bhunger = max(user.bhunger-2*drain,0)
-	if(drain>=50)
-		user.visible_message("<span class='danger'>[user]'s eyes give off eerie red glow!</span>", \
-		"<span class='danger'>...but it wasn't nearly enough. You crave, crave for more. The hunger consumes you from within.</span>", \
-		"<span class='italics'>You hear a heartbeat.</span>")
-		user.bhunger += drain
-		src = user
-		spawn()
-			for (,user.bhunger>0,user.bhunger--)
-				sleep(50)
-				user.take_overall_damage(3, 0)
+		offering = possible_targets[possible_targets.len]
+	if(!offering)
 		return
-	user.heal_organ_damage(drain%5, 0)
-	drain-=drain%5
-	for (,drain>0,drain-=5)
-		sleep(2)
-		user.heal_organ_damage(5, 0)
-	return
-
-
-///Raise Dead: Requires two corpses. Brings one to life and gibs the other.
-
-
-/obj/effect/rune/proc/raise()
-	var/mob/living/carbon/human/corpse_to_raise
-	var/mob/living/carbon/human/body_to_sacrifice
+	if(offering.null_rod_check())
+		user << "<span class='warning'>Something is blocking the Geometer's magic!</span>"
+		log_game("Sacrifice rune failed - target has null rod")
+		fail_invoke()
+		return
+	if(ishuman(offering) || isrobot(offering))
+		var/cultists_nearby = 1
+		for(var/mob/living/M in orange(1,src))
+			if(iscultist(M) && M != user)
+				cultists_nearby++
+				M.say(invocation)
+		if(cultists_nearby < 3)
+			user << "<span class='warning'>You require three acolytes to sacrifice greater targets!</span>"
+			fail_invoke()
+			log_game("Sacrifice rune failed - not enough acolytes and target is living")
+			return
+	visible_message("<span class='warning'>[src] pulses blood red!</span>")
+	sac(offering)
 
-	var/is_sacrifice_target = 0
-	for(var/mob/living/carbon/human/M in src.loc)
-		if(M.stat == DEAD)
-			if(ticker.mode.name == "cult" && M.mind == ticker.mode:sacrifice_target)
-				is_sacrifice_target = 1
-			else
-				corpse_to_raise = M
-				if(M.key)
-					M.ghostize(1)	//kick them out of their body
-				break
-	if(!corpse_to_raise)
-		if(is_sacrifice_target)
-			usr << "<span class='danger'>The Geometer of blood wants this mortal for himself.</span>"
-		return fizzle()
-
-
-	is_sacrifice_target = 0
-	find_sacrifice:
-		for(var/obj/effect/rune/R in world)
-			if(R.word1==wordblood && R.word2==wordjoin && R.word3==wordhell)
-				for(var/mob/living/carbon/human/N in R.loc)
-					if(ticker.mode.name == "cult" && N.mind && N.mind == ticker.mode:sacrifice_target)
-						is_sacrifice_target = 1
+/obj/effect/rune/sacrifice/proc/sac(mob/living/T)
+	var/sacrifice_fulfilled
+	if(T)
+		if(T.mind)
+			var/obj/item/device/soulstone/stone = new /obj/item/device/soulstone(get_turf(src))
+			if(!stone.transfer_soul("FORCE", T, usr)) //If it cannot be added
+				qdel(stone)

```

Comment:
there's a sleep in transfer_soul(), which means you can spamclick the rune to make appear multiples shards (which will be destroyed...probably). And this can cause a runtime since we don't check if(T) again after the transfer_soul sleep.